### PR TITLE
Build for mac and winodws [DEVINFRA-768]

### PR DESCRIPTION
**Pull request by adrian-kong**
Builds other releases

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      - name: Build
+        shell: bash
+        run: |
+          cargo make --profile ${{ matrix.profile }} build
+
+#      - name: Package Release
+#        shell: bash
+#        run: |
+#          touch LICENSE.txt
+#          cargo make zip-release-binary-for-target \
```
Comment (adrian-kong) [inline .github/workflows/main.yml]: using https://github.com/sagiegurari/cargo-make/blob/master/docs/cargo_make_task_list.md
but it gets skipped locally - could just upload the esthri file?

File: Makefile.toml [hunk 2]
```diff
@@ -14,6 +14,14 @@ BUILD_FLAVOR = "--release"
 BUILD_FLAVOR = "--release"
 TARGET = "--target=x86_64-unknown-linux-musl"
 
+[env.'release+mac']
+BUILD_FLAVOR = "--release"
+TARGET = "--target=x86_64-apple-darwin"
```
Comment (adrian-kong) [inline Makefile.toml:19]: should use `x86_64` instead of `aarch64`?

Comment (samvrlewis) [reply Makefile.toml:19]: Both, if possible, so we can support both x86 and M1 macs

File: .github/workflows/main.yml (hunk 1 above)
Comment (samvrlewis) [reply .github/workflows/main.yml]: You could try to figure out why its getting skipped (suspect its something to do with the multi-crate setup) with `cargo make --verbose zip-release-binary-for-target` or maybe, yeah, we could just add our own commands to build + zip

**Comment by adrian-kong** [comment]
reopened as #305 because of some error i had `dst refspec refs matches more than one`

**Pull request closed without merging**